engine/schema/src/main/java/com/cloud/upgrade/GuestOsMapper.java

```diff
@@ -188,4 +190,33 @@ private boolean isValidGuestOSHypervisorMapping(GuestOSHypervisorMapping mapping
         LOG.warn("Invalid Guest OS hypervisor mapping");
         return false;
     }
+
+    /**
+     * Copies guest OS mappings from src version to dest version for the hypervisor (use this to copy all mappings from older version to newer version during upgrade)
+     * @return true if copied successfully, else false.
+     */
+    public boolean copyGuestOSHypervisorMappings(HypervisorType hypervisorType, String srcVersion, String destVersion) {
+        if (hypervisorType == HypervisorType.None || hypervisorType == HypervisorType.Any) {
+            LOG.warn("Unable to copy, invalid hypervisor");
+            return false;
+        }
+
+        if (StringUtils.isBlank(srcVersion) || StringUtils.isBlank(destVersion)) {
```

Comment:
```suggestion
        if (StringUtils.isAnyBlank(srcVersion, destVersion)) {
```

Reply:
@harikrishna-patnala why not apply this?